.github/workflows/ci/pkg/environment/environment.go

```diff
@@ -111,22 +116,75 @@ func New(c Config) (*PullRequestEnvironment, error) {
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	// Check if the pull request has changes to the `docs/` directory as that will effect who
+	// the required reviewers are.
+	docChanges, err := hasDocChanges(c.Context, pr, c.Client)
+	if err != nil {
+		// Log the error, don't fail the whole run because it couldn't detect if the pull request has docs
+		log.Errorf("error while detecting pull request for docs changes: %v, skipping assigning docs reviewers", err)
+	}
+
 	return &PullRequestEnvironment{
 		Client:           c.Client,
 		reviewers:        c.Reviewers,
 		defaultReviewers: c.Reviewers[ci.AnyAuthor],
 		Metadata:         pr,
+		HasDocsChanges:   docChanges,
 	}, nil
 }
 
+// hasDocsChanges determines if the pull request has changes to `docs/`
+func hasDocChanges(ctx context.Context, pr *Metadata, clt *github.Client) (bool, error) {
+	files, err := getPullRequestFiles(ctx, pr, clt)
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+	for _, file := range files {
+		if file.Filename == nil {
+			return false, trace.BadParameter("pull request file name is nil")
+		}
+		if strings.HasPrefix(*file.Filename, ci.DocsPrefix) {
+			return true, nil
+		}
+	}
+	return false, nil
+}
```

Comment:
I think you need two bits of information here to correctly assign reviewers: does the PR have docs changes and if the PR has code changes.

So I would suggest you update this function (and rename) so it returns two bools: `hasDocsChanges` and `hasCodeChanges`.

* `hasDocsChanges` is true if any files under `docs/` have been updated.
* `hasCodeChanges` if changes are anywhere else.

Then `GetReviewersForAuthor` simply becomes a truth table:

```go
func (e *PullRequestEnvironment) GetReviewersForAuthor(user string) []string {
  var reviewers []string

  switch {
    case docsChanges && codeChanges:
      reviewers = append(reviewers, docsReviewers)
      reviewers = append(reviewers, codeReviewers)
    case docsChanges  && !codeChanges:
      reviewers = append(reviewers, docsReviewers)
    case !docsChanges && codeChanges:
      reviewers = append(reviewers, codeReviewers)
    // Strange state.
    case !docsChanges && !codeChanges:
      reviewers = append(reviewers, codeReviewers)
}
  return reviewers
}

Reply:
Just curious, in what case would someone open an empty PR? If someone opened an empty PR in your scenario, reviewers still get assigned (my logic also does the same).

Reply:
I don't know either, which is why I called it a strange state. However, if there is a need to create an empty PR, we shouldn't bother docs reviewers with it.